Return type of the incorrect value in EnsureError

KeyWeeUsr opened the pull request:
Thanks for this package, I love it! However it misses a quite important info which is present in most of the compilers I've encountered (not in all places unfortunately) and that is the type/class of the original value instead of the target value (annotation).

Why this is a problem - imagine e.g. compiling C or Java and getting a compiler error just saying "sorry, I don't accept this type". While in low-level languages it might be enough e.g. due to limitation of not being able to determine the original value type or just sparing some CPU cycles during compilation, it's way better to have explicitly said what type I tried to push through that annotation mainly for dynamically typed languages.

In small codebases it might not be important enough, however the more complex the code gets the better is for a dynamically typed language to have explicitly said what is the wrong original/inputted type instead of manually tracing through multiple files and guessing where let's say a `unicode`/`str` changed to `bytes` or `list` into `tuple` or even some object into `None`.

kislyuk commented:
This is a great addition, thank you. Can you please fix the tests? It looks ready to go otherwise.

KeyWeeUsr commented:
Fixed. Also I've fixed the regex for methods because it failed for me and apparently there is a difference between GNU/Linux distros (lowercase hex) and Windows (uppercase hex).

kislyuk commented:
Released in v0.8.1.
